### ISSUE TITLE
document file listing behaviour more comprehensible

### ISSUE THREAD

**Issue by RonnyPfannschmidt** [pypa/setuptools-scm#175]
the small mention of file listers doesn't help people unaware of the setuptools feature to even grasp its importance or use it as intended

more detailed explanations and examples are needed

**Comment by antocuni** [comment]
Also, from the point of view of someone who writes `setup.py` but it's not particularly aware of all the distutils/setuptools quirks, it is a bit unclear what is the relationship of `setuptools_scm` w.r.t. `package_data`, `include_package_data` and `MANIFEST.in`

**Comment by gaborbernat** [comment]
so can someone explain? :+1: from what I see ``setuptools_scm`` forces everything to be packaged, which would make package_data redundant

**Comment by RonnyPfannschmidt** [comment]
@gaborbernat bascially setuptools_scm also registers a file lister for git/hg, thus adding all files to the sdists

**Comment by gaborbernat** [comment]
yeah, I saw that reading through the code, I see how some people probably really on that, but it completely messes up people using the canonical ``find_packages``, and ``package_data`` mechanism - maybe that setuptools file lister should be opt out able! thanks :+1:

**Comment by RonnyPfannschmidt** [comment]
@gaborbernat due to setuptools limitations its impossible to easily opt out as its an api thats kept completely distinct

i'm willing to accept a clean implementation of a sane opt out, i'm not going to invent one and i'm going to reject messy ones

**Comment by jaraco** [comment]
In the past, setuptools had Subversion support built-in. If you had a subversion repo, it presumed you wanted to use the repo to manage your package contents. Setuptools_scm builds on this assumption, but around git and hg.

But you bring up a good point - that you may want the versioning support but want to disregard the file finder support.

In my opinion, the file finder support is superior and should be preferred if available, but I'm not yet convinced that there's a case where the presence of a file finder is undesirable.

If there is such a case that can be made, it's probably in Setuptools that the functionality should be added so that it can apply to other file finders (setuptools_svn, future SCMs, or alternates).

**Comment by gaborbernat** [comment]
@jaraco I agree, looking at the bits and codes, the setuptools exclude should apply to SCM code and not taken blindly 👍 on the other hand it's interesting how should file finders interact with ``find_packages`` and ``data_files`` settings.

**Comment by RonnyPfannschmidt** [comment]
recent documentation updates resolved this
